### ISSUE TITLE
Running react-navigation/stack on Expo 49 Production Mode crashes.

### ISSUE THREAD

**Issue by edumorlom** [react-navigation/react-navigation#11494]
### Current behavior

When I run the following on Expo 49 production mode, the app crashes. Expo 48 is OK.
The weird thing is that if I run it on dev mode, it works fine. Something is breaking in production mode.
The code is very simple to reproduce.

`npx expo start --no-dev --tunnel`

```
import 'react-native-gesture-handler';
import {createStackNavigator} from '@react-navigation/stack';

function App() {
  return <></>
}
```

### Expected behavior

The app should not crash.

### Reproduction

.

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

"@expo/ngrok": "^4.1.0",
    "@react-native-async-storage/async-storage": "1.19.1",
    "@react-native-community/cli-server-api": "^9.2.1",
    "@react-native-community/datetimepicker": "7.2.0",
    "@react-native-community/masked-view": "0.1.10",
    "@react-native-community/picker": "1.6.6",
    "@react-native-picker/picker": "2.4.10",
    "@react-navigation/native": "6.1.7",
    "@react-navigation/stack": "6.3.17",
    "eas-cli": "^3.17.0",
    "expo": "^49.0.5",
    "expo-calendar": "~11.3.0",
    "expo-constants": "~14.4.2",
    "expo-haptics": "~12.4.0",
    "expo-image-picker": "~14.3.2",
    "expo-localization": "~14.3.0",
    "expo-location": "~16.1.0",
    "expo-notifications": "~0.20.1",
    "expo-permissions": "~14.2.1",
    "expo-updates": "~0.18.11",
    "firebase": "^9.3.0",
    "firebase-admin": "^9.3.0",
    "firebase-functions": "^3.24.1",
    "geolib": "^3.3.1",
    "moment": "^2.29.4",
    "paper": "0.12.8",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.72.3",
    "react-native-confetti-cannon": "^1.5.2",
    "react-native-dialog-input": "^1.0.8",
    "react-native-dropdown-picker": "^3.7.1",
    "react-native-elements": "^3.0.0",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-haptic-feedback": "^1.11.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.3",
    "react-native-maps": "1.7.1",
    "react-native-masked-text": "^1.13.0",
    "react-native-material-dropdown-v2": "^0.12.0",
    "react-native-modal-datetime-picker": "^15.0.0",
    "react-native-paper": "^4.12.5",
    "react-native-reanimated": "~3.3.0",
    "react-native-responsive-fontsize": "^0.5.1",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-swipe-gestures": "^1.0.5",
    "react-native-vector-icons": "^9.2.0",
    "expo-document-picker": "~11.5.3"

**Comment by connor-wayne-mcelroy** [comment]
klh

**Comment by adamtsai1** [comment]
@edumorlom We just burned 2 days trying to track this one down. For us, removing react-native-reanimated fixed the crashing. Thankfully it didn't appear as though it was being used anywhere, but a thorough test is still on deck.

**Comment by ddehghan** [comment]
I think this also has to do with my bug: https://github.com/react-navigation/react-navigation/issues/11494

react-native-reanimated had a crash bug a months ago but they fixed that and in this version should be working "3.5.4",

**Comment by edumorlom** [comment]
Yep, it has been fixed! Thanks for letting me know @ddehghan 

Really appreciated, wasnt aware. It works fine now. Closing.

**Comment by ddehghan** [comment]
Wait a minute. Why did you close my bug? This is still happening. Can you tell me where it has been fixed. it is already release? or can you point to PR?

**Comment by ddehghan** [comment]
Wait a minute. Why did you close my bug? This is still happening. Can you tell me where it has been fixed. it is already released? or can you point to PR? or a merge or something

The bug that I filed is with the latest lib version.

**Comment by ButuzGOL** [comment]
when i add "react-native-reanimated": "3.5.4",
it throws Error: [Reanimated] Mismatch between JavaScript part and native part of Reanimated (3.5.4 vs 3.3.0).

**Comment by ButuzGOL** [comment]
yes i am using expo

**Comment by ButuzGOL** [comment]
expo link https://github.com/expo/expo/issues/24828

**Comment by lsps9150414** [comment]
@ButuzGOL have you fix this by upgrading to `3.5.4`? I'm still having the issue with `3.5.4`. I'm using `expo-router` which I'm not sure yet if is related.
